src/Symfony/Component/Validator/Constraints/When.php

```diff
@@ -35,13 +36,16 @@ class When extends Composite
      * @param string[]|null                         $groups
      * @param array<string,mixed>                   $options
      */
+    #[HasNamedArguments]
     public function __construct(string|Expression|array $expression, array|Constraint|null $constraints = null, ?array $values = null, ?array $groups = null, $payload = null, array $options = [])
```

Comment:
the options argument also needs to be deprecated

Reply:
fixed